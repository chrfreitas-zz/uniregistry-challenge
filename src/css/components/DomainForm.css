
.domain-form {
    transition: opacity .5s, transform .5s;
    padding: 15px;
    border-radius: 5px;
    opacity: 0;
    top: 50%;
    background: rgba(0, 0, 0, 0.2);
    color: #fff;
    transform: scale(.8);
}

.is-editing {
    .domain-form {
        opacity: 1;
        transform: scale(1);
    }
}