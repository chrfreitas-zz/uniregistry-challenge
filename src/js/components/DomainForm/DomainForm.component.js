import React, { Component } from 'react';
import PropTypes from 'prop-types';

class DomainForm extends Component {
    static propTypes = {
        domain: PropTypes.object,
        update: PropTypes.func
    }

    state = {
        domain: this.props.domain
    }

    handleChange = (event) => {
        const newState = {
            domain: {
                ...this.state.domain,
                [event.target.name]: event.target.value
            }
        }

        this.setState(newState);
    }

    update = (e) => {
        e.preventDefault();
        this.props.update(this.state.domain);
    }

    render() {
        return (
            <div>
                <form>
                    <div>
                        <label>Domain name</label>
                        <input type="text" name="name" value={this.state.domain.name} onChange={this.handleChange} />
                    </div>
                    <div>
                        <label>Registran email</label>
                        <input type="text" name="email" value={this.state.domain.email} onChange={this.handleChange} />
                    </div>
                    <div>
                        <label>Price</label>
                        <input type="text" name="price" value={this.state.domain.price} onChange={this.handleChange} />
                    </div>
                    <div>
                        <button onClick={this.update}>Update it!</button>
                        <a href="">Cancel</a>
                    </div>
                </form>
            </div>
        );
    }
}

export default DomainForm;