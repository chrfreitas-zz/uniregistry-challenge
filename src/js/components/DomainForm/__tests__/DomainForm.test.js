import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import DomainForm from '../DomainForm.component';

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

describe('DomainForm component', () => {
    let domain = {};
    beforeAll(() => {
        domain = {
            id: 2,
            name: 'lalala',
            email: 'lalala@gmail.com',
            price: '$3.00'
        }
    });

    it('should match with the snapshot', () => {
        const wrapper = renderer.create(<DomainForm domain={domain}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    describe('handleChange()', () => {
        it('should update values in domain state', () => {
            const event = {
                target: {
                    name: 'description',
                    value: 'chrfreitas.com'
                }
            }

            const wrapper = shallow(<DomainForm domain={domain}/>);
            wrapper.find('[name="description"]').simulate('change', event);

            expect(wrapper.state().domain.description).toBe(event.target.value);
        });
    })
});