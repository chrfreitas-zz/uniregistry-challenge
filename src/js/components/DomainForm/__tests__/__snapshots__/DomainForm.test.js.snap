// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainForm component should match with the snapshot 1`] = `
<form>
  <div
    className="form-group"
  >
    <label>
      Domain name
    </label>
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Registran email
    </label>
    <input
      className="form-control"
      name="email"
      onChange={[Function]}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Price
    </label>
    <input
      className="form-control"
      name="price"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Update it!
    </button>
    <a
      className="btn btn-link"
      onClick={[Function]}
    >
      Cancel
    </a>
  </div>
</form>
`;
